code/go/0chain.net/smartcontract/interestpoolsc/sc_test.go

```diff
@@ -0,0 +1,784 @@
+				inputData: testPoolState().encode(),
+				balances:  testBalance(clientID1, 10),
+			},
+			wantErr: true,
+			want:    "",
+		},
+		{
+			name: "ok",
+			args: args{
+				t:         testTxn(clientID1, 10),
+				un:        testUserNode(clientID1, testInterestPool(0, 100)),
+				gn:        testGlobalNode(globalNode1Ok, 100, 1, 100, 0, 10*time.Second),
+				inputData: testPoolState().encode(),
+				balances:  testBalance(clientID1, 10),
+			},
+			wantErr:    false,
+			want:       "",
+			shouldBeOk: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ip := &InterestPoolSmartContract{
+				SmartContract: tt.fields.SmartContract,
+			}
+			sc := &smartcontractinterface.SmartContract{
+				RestHandlers:                map[string]smartcontractinterface.SmartContractRestHandler{},
+				SmartContractExecutionStats: map[string]interface{}{},
+			}
+			ip.SetSC(sc, nil)
+			got, err := ip.unlock(tt.args.t, tt.args.un, tt.args.gn, tt.args.inputData, tt.args.balances)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("unlock() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if tt.shouldBeOk {
+				tpr := &tokenpool.TokenPoolTransferResponse{
+					ToClient: clientID1,
+					Value:    100,
+					FromPool: "new_test_pool_state"}
+				tt.want = string(tpr.Encode())
+			}
+			if got != tt.want {
+				t.Errorf("unlock() got = %v, want %v", got, tt.want)
+			}
+
+		})
+	}
+}
+
+func TestInterestPoolSmartContract_updateVariables(t *testing.T) {
+	type fields struct {
+		SmartContract *smartcontractinterface.SmartContract
+	}
+	type args struct {
+		t         *transaction.Transaction
+		gn        *GlobalNode
+		inputData []byte
+		balances  state.StateContextI
+	}
+	tests := []struct {
+		name       string
+		fields     fields
+		args       args
+		want       string
+		wantErr    bool
+		shouldBeOk bool
+	}{
+		{
+			name: "unauthorized access",
+			args: args{
+				t:         testTxn(clientID1, 100),
+				gn:        nil,
+				inputData: nil,
+				balances:  nil,
+			},
+			want:    "",
+			wantErr: true,
+		},
+		{
+			name: "request not formatted correctly",
+			args: args{
+				t:         testTxn(owner, 100),
+				gn:        nil,
+				inputData: []byte("{test}"),
+				balances:  nil,
+			},
+			want:    "",
+			wantErr: true,
+		},
+		{
+			name: "ok",
+			args: args{
+				t:         testTxn(owner, 100),
+				gn:        testGlobalNode(globalNode1Ok, 10, 10, 0, 10, 5),
+				inputData: testGlobalNode(globalNode1Ok, 10, 20, 30, 40, 10).Encode(),
+				balances:  testBalance("", 0),
+			},
+			want:       string(testGlobalNode(globalNode1Ok, 10, 10, 30, 40, 10).Encode()),
+			wantErr:    false,
+			shouldBeOk: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ip := &InterestPoolSmartContract{
+				SmartContract: tt.fields.SmartContract,
+			}
+			if tt.shouldBeOk {
+				config.SetupSmartContractConfig()
```

Comment:
TestInterestPoolSmartContract_updateVariables fails because `config.SetupSmartContractConfig` is expecting a `config` directory with a `sc.???` file in, which is missing.

Elsewhere `config.SetupSmartContractConfig` is only used in production code. This needs to be mocked somehow.

Reply:
Why do you want to mock config? this does not change the state of the application, it only sets the input parameters

Reply:
TestInterestPoolSmartContract_updateVariables fails because `config.SetupSmartContractConfig` is expecting a `config` directory with a `sc.???` file in, which is missing.

Some change is needed to stop the unit test from failing at this point.

I do not want a mock config. There just needs to be some change so the unit test works.  A mock config looked like one possible solution to me, but whatever works.